.github/actions/oblt-cli-destroy-cluster/action.yml

```diff
@@ -4,15 +4,53 @@ inputs:
   cluster-name:
     description: 'The cluster name'
     required: false
+  cluster-info-file:
+    description: 'The cluster info file (absolute path)'
+    required: false
   github-token:
     description: 'The GitHub access token.'
     required: true
 runs:
   using: "composite"
   steps:
+    - name: Process inputs
+      uses: actions/github-script@v6
+      id: process
+      env:
+        CLUSTER_NAME_INPUT: ${{ inputs.cluster-name }}
+        CLUSTER_INFO_FILE_INPUT: ${{ inputs.cluster-info-file }}
+      with:
+        script: |
+          const clusterNameInput = process.env.CLUSTER_NAME_INPUT
+          const clusterInfoFile = process.env.CLUSTER_INFO_FILE_INPUT
+
+          if (clusterNameInput && clusterInfoFile) {
+            core.setFailed(`Either cluster-name or cluster-info-file but no both`)
+            return
+          }
+
+          var clusterName = clusterNameInput
+
+          if (clusterInfoFile) {
+            const fs = require('fs')
+            const data = fs.readFileSync(clusterInfoFile, 'utf8')
+            const jsonObject = JSON.parse(data)
+            if (jsonObject && jsonObject.ClusterName === undefined) {
+              core.setFailed(`cluster-info-file does not contain the field ClusterName`)
+              return
+            }
+            clusterName = jsonObject.ClusterName
+          }
+
+          if (clusterName.trim().length === 0) {
+            core.setFailed(`cluster name is empty`)
+            return
+          }
+
+          console.log(`clusterName is "${clusterName}"`)
+          core.setOutput('clusterName', `${clusterName}`)
+
```

Comment:
NIT: we can extract this to a new action to reuse it in more places

Reply:
💯 , let me do it

Reply:
see https://github.com/elastic/apm-pipeline-library/pull/2428/commits/a51922f3121610f037f162bd16b56a28d596ee93